.github/workflows/README.md

```diff
@@ -0,0 +1,23 @@
+# Github Actions Workflows
+
+## Benchmarking Performance Tests
+`benchmarks.yml` contains a workflow to check for any performance regressions or
+improvements in benchmark tests. 
+
+It uses
+[github-action-benchmark](https://github.com/benchmark-action/github-action-benchmark)
+to check performance diffs between a PR and the `master` branch, comments if it
+there is a regression past a certain threshold (default: `200%`), and generates
+a performance diff JSON between consecutive commits in the `master` branch in
+the `gh-pages` branch (the JSON is visualized into a graph that can be seen at:
+https://algorand.github.io/go-algorand/dev/bench/).
+
+### Adding benchmark tests
+Currently, the workflow runs the `BenchmarkUintMath` in the `Run benchmark`
+step. Additional benchmarks can be run using the `-bench` flag. 
+
+### CI Variance
+There may be some variance between runs because github actions might spin up a
+different machine each time (e.g. Intel Xeon 8370C vs 8171M; the latter might
+run benchmarks slightly slower). Empirically, the variance seems to be 10~30%
+for the most part. 
```

Comment:
@algochoi Minor and optional:  As we've discussed verbally, in its current form, the workflow is best suited for finding large performance degradations rather than minute changes.

Optionally, I propose adding a sentence explicitly highlighting the point to the reader.

```suggestion
for the most part. 

Due to the environment variance, the workflow is most suitable for finding _large_ performance degradations.
```